tests/test_stubs.py

```diff
@@ -1,34 +1,60 @@
-import os.path
+from pathlib import Path
 import pytest
 from mypy import api
 
 
-TESTS_DIR = os.path.dirname(__file__)
+TESTS_DIR = Path(__file__).parent
 
 
 def gen_tests():
-    for filename in os.listdir(TESTS_DIR):
-        if filename.endswith('.py') and not filename.startswith('test_'):
+    '''List of all tests files included in the directory tests'''
+    for filename in TESTS_DIR.glob('*.py'):
+        if not str(filename.parts[-1]).startswith('test_'):
             yield filename
 
+def gen_abs_qflags_tests():
+    '''List of all tests included in the directory qflags'''
+    for filename in (TESTS_DIR/'qflags').glob('test_*.py'):
+        yield filename
 
-@pytest.mark.parametrize('filename', list(gen_tests()))
-def test_stubs(filename):
+
+@pytest.mark.parametrize('filename',
+                         list(gen_tests()),
+                         ids=[v.parts[-1] for v in gen_tests()]
```

Comment:
Could be `.name` but this avoids prescribing a single directory being handled.  Also, `.as_posix()` will give a `str` with consistent forward slashes.
```suggestion
                         ids=[v.relative_to(TESTS_DIR).as_posix() for v in gen_tests()]
```